be/src/olap/tablet_manager.cpp

```diff
@@ -172,24 +183,29 @@ OLAPStatus TabletManager::_add_tablet_to_map_unlocked(TTabletId tablet_id, Schem
     RETURN_NOT_OK_LOG(tablet->register_tablet_into_dir(), Substitute(
             "fail to register tablet into StorageEngine. data_dir=$0",
             tablet->data_dir()->path()));
-
-    _tablet_map[tablet_id].table_arr.push_back(tablet);
-    _tablet_map[tablet_id].table_arr.sort(_cmp_tablet_by_create_time);
-    _partition_tablet_map[tablet->partition_id()].insert(tablet->get_tablet_info());
+    tablet_map_t& tablet_map = _get_tablet_map(tablet_id);
+    tablet_map[tablet_id].table_arr.push_back(tablet);
+    tablet_map[tablet_id].table_arr.sort(_cmp_tablet_by_create_time);
+    {
+        ReadLock rlock(&_partition_tablet_map_lock);
+        _partition_tablet_map[tablet->partition_id()].insert(tablet->get_tablet_info());
```

Comment:
This method name is `_add_tablet_to_map_unlocked()`, why you add a rlock here?

Reply:
here should be wlock， to protect _partition_tablet_map，_add_tablet_to_map_unlocked is that thread not obtained _tablet_map_lock